main.go

```diff
@@ -45,6 +46,32 @@ func main() {
 	arn := ""
 	var err error
 
+	if *requireLatest {
+		// check if the latest task definition is running. if it is not, do not proceed with the deployment.
+		taskDef, err := c.GetTaskDefinition(task)
+		if err != nil {
+			logger.Printf("[error] get task definition: %s\n", err)
+			os.Exit(1)
+		}
+
+		deployments, err := c.GetDeployments(cluster, service)
+		// there can be more than 1 deployment running if a deployment is in progress,
+		// not worth handling at this point: it's not possible to determine if the latest deployment
+		// will succeed, for example.
+		if len(deployments) != 1 {
+			logger.Printf("[error] not exactly one deployment found: %d\n", len(deployments))
+			os.Exit(1)
+		}
+
+		d := deployments[0]
+		if *d.TaskDefinition != *taskDef.TaskDefinitionArn {
+			logger.Printf("[error] latest task definition not running: %s\n", *d.TaskDefinition)
```

Comment:
It may be helpful to add something a little more descriptive for this error like how you may need to deregister latest task definitions.

Reply:
i'm updating the runbook. if the latest task definition isn't running, the next indicated step isn't known at the point of this error. it may be to deregister, it may be to roll forward.